lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -455,24 +455,31 @@ def update_positions
 
           def internal_scope_changed?
             return @scope_changed if defined?(@scope_changed)
-            
+
             @scope_changed = scope_changed?
           end
 
           # Temporarily swap changes attributes with current attributes
-          def swap_changed_attributes
-            @changed_attributes.each do |k, _|
-              if self.class.column_names.include? k
-                @changed_attributes[k], self[k] = self[k], @changed_attributes[k]
-              end
+          def with_old_attributes
+            begin
+              changed_attributes.each { |k, _| @changed_attributes[k], self[k] = self[k], @changed_attributes[k] }
+              yield
+            ensure
+              changed_attributes.each { |k, _| @changed_attributes[k], self[k] = self[k], @changed_attributes[k] }
             end
           end
 
           def check_scope
             if internal_scope_changed?
-              swap_changed_attributes
-              send('decrement_positions_on_lower_items') if lower_item
-              swap_changed_attributes
+              with_old_attributes do
+                # if position unchanged and db has a different value for position column
+                unless send("#{position_column}_changed?")
```

Comment:
This won't be true anymore if you use my method of swapping attributes unfortunately.